components/BottomNavigation/src/private/MDCBottomNavigationLargeItemDialogView.m

```diff
@@ -0,0 +1,133 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBottomNavigationLargeItemDialogView.h"
+
+#import <CoreGraphics/CoreGraphics.h>
+
+static const CGFloat kTitleFontScaling = (CGFloat)0.5;
+static const CGFloat kTitleFontSize = 35;
+static const CGFloat kImageViewHeightToDialogRatio = (CGFloat)0.35;
+
+/**
+ * Returns an image representing the given UITabBarItem. If the item has a largeContentSizeImage,
+ * this function will return that image, otherwise it returns the value of the item's image
+ * property. This value may be nil.
+ */
+static UIImage *_Nullable MDCImageForItem(UITabBarItem *_Nullable item) {
+  if (!item) {
+    return nil;
+  }
+
+  UIImage *image;
+  if (@available(iOS 11, *)) {
+    image = item.largeContentSizeImage;
+  }
+
+  return image ?: item.image;
+}
+
+@interface MDCBottomNavigationLargeItemDialogView ()
+
+/** The view that hosts the image of the tab bar item. */
+@property(nonatomic, nonnull) UIImageView *imageView;
+
+/** The view that is set to the title of the tab bar item. */
+@property(nonatomic, nonnull) UILabel *titleLabel;
+
+@end
+
+@implementation MDCBottomNavigationLargeItemDialogView
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonInitMDCBottomNavigationLargeItemView];
+  }
+
+  return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonInitMDCBottomNavigationLargeItemView];
+  }
+
+  return self;
+}
+
+- (void)commonInitMDCBottomNavigationLargeItemView {
+  UIColor *contentColor = [UIColor colorWithWhite:(CGFloat)0.15 alpha:1];
+
+  _imageView = [[UIImageView alloc] init];
+  _imageView.tintColor = contentColor;
+  _imageView.contentMode = UIViewContentModeScaleAspectFit;
+  [self.contentView addSubview:_imageView];
+
+  _titleLabel = [[UILabel alloc] init];
+  _titleLabel.textAlignment = NSTextAlignmentCenter;
+  _titleLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+  _titleLabel.adjustsFontSizeToFitWidth = YES;
+  _titleLabel.minimumScaleFactor = kTitleFontScaling;
+  _titleLabel.textColor = contentColor;
+  _titleLabel.font = [UIFont systemFontOfSize:kTitleFontSize];
+  _titleLabel.numberOfLines = (_imageView.image) ? 1 : 0;
+  [self.contentView addSubview:_titleLabel];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  UIEdgeInsets margins = self.layoutMargins;
+  CGFloat dialogHeight = MAX(0, CGRectGetHeight(self.bounds) - margins.top - margins.bottom);
+  CGFloat dialogWidth = MAX(0, CGRectGetWidth(self.bounds) - margins.left - margins.right);
+  CGFloat imageHeight =
+      self.imageView.image ? (CGFloat)floor(dialogHeight * kImageViewHeightToDialogRatio) : 0;
```

Comment:
Tip: I'd prefer to see a fixed height for the image, consistent across devices.  If the view has additional vertical space, I think it's worth dedicating it to the label where possible.

Reply:
This also doesn't seem to match the behavior of UITabBarController in my testing.  The image below shows a blown-up version with the vertical midpoint identified with the dotted line.

![Screen Shot 2019-04-16 at 8 42 44 AM](https://user-images.githubusercontent.com/1753199/56210453-dd093200-6023-11e9-9eed-0e8791f8c8fa.png)

Reply:
Done.  Changed the image to a fixed height and positioning to a fixed offset from the top of the dialog.

Reply:
<img width="272" alt="Screen Shot 2019-04-16 at 17 30 53" src="https://user-images.githubusercontent.com/3827017/56252632-8294c380-606d-11e9-9440-ae453cd8fc35.png">